misk-crypto/README.md

```diff
@@ -0,0 +1,59 @@
+This module provides Misk with the ability to store and manage cryptographic key material in a
+safe and easy way.
+
+How it works?
+-----
+The `Cipher` object is used to encrypt/decrypt data.
+
+It uses a key which is loaded from a configuration file at startup.
+The key associated with each cipher should be stored **encrypted** in the app's common configuration file.
```

Comment:
Does this live on the same host as the app? Does the app need to send the key to another host or service to decrypt it? That was a goal we had with the internal predecessor .

Reply:
I'm not sure I understand. Are you referring to how the key is being decrypted?
If so, then the way it works is:
- When the app loads, it instantiates a `Aead` object via the KMS.
- Then, for each encrypted key in the configuration file:
  - Decrypt the key using the above `Aead` and create a new `Cipher` object with the decrypted result.

The key point here is that the decrypted key is only available in memory and is not being written anywhere to disk.

Reply:
Makes sense. I think the important thing is that if someone were to get access to the data on disk, they wouldn't be able to decrypt the encrypted data (because they don't have access to the KMS.